roles/users/tasks/main.yml

```diff
@@ -50,7 +50,7 @@
   changed_when: false
   sudo: no
   register: admin_user_status
-  when: not sshd_permit_root_login
+  when: not sshd_permit_root_login and ansible_ssh_host is defined
```

Comment:
For clarity (this was confusing to me initially): Ansible's `--connection=local` flag means that the playbook is run on the local machine (i.e., on your Mac, were you so inclined). There is no SSH involved at all. In the context of Packer, this means that the playbooks, roles, group_vars, host_vars, etc., are uploaded to the guest machine (whether local VM or cloud) and run on the guest with `--connection=local`. This task fails on Packer because we are not connecting via SSH.

Huge thanks for pointing out that defining hosts by IP means that `ansible_ssh_host` will be undefined. That was unexpected! And, _of course_, a different search term of the docs turns up `ansible_connection`. It is defined when the `--connection` flag is passed. Sheesh. I'll push an update.